Assets/Scripts/Basic Player Movement/CharacterController.cs

```diff
@@ -0,0 +1,56 @@
+using UnityEngine;
+
+public class CharacterController : MonoBehaviour
+{
+    [SerializeField]
+    private float speed = 6f; // Speed of the character
+    [SerializeField]
+    private float jumpForce = 5f; // Force of the character's jump
+    [SerializeField]
+    private float rotationSpeed = 80f; // Rotation speed of the character
+    [SerializeField]
+    private Transform groundCheck; // Transform used to check if the character is grounded
+
+    private Rigidbody rb; // Rigidbody component of the character
+    private bool isGrounded; // Flag to track if the character is grounded
+
+    void Start()
+    {
+        // Get the Rigidbody component
+        rb = GetComponent<Rigidbody>();
+    }
+
+    void Update()
+    {
+        // Check if the character is grounded
+        isGrounded = Physics.CheckSphere(groundCheck.position, 0.1f);
+
+        // Get the horizontal and vertical input from the keyboard
+        float horizontalInput = Input.GetAxis("LookHorizontal");
+        float horizontalMoveInput = Input.GetAxis("Horizontal");
+        float verticalInput = Input.GetAxis("Vertical");
```

Comment:
Would rename these. Not fussed though since this is a throwaway script but worth thinking about

Reply:
that was my laziness. I was meant to rename them appropriately but didn't